### PR TITLE
8295158: G1: Increase card-based cost sample reporting threshold

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  can I have reviews for this change that makes the threshold that determines whether g1 uses that sample for prediction configurable (as diagnostic option) and increases it to 1000 instead of 10 because testing has shown that that's a much better value, filtering out outliers much better.

This change is based on PR#10654, please review that one first if still open.

Testing: Testing: gha, local testing, tier1-5 with other changes in this patch series, perf testing with other changes in this patch series

Thanks,
  Thomas

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295158](https://bugs.openjdk.org/browse/JDK-8295158): G1: Increase card-based cost sample reporting threshold

### Reviewers
 * [Ivan Walulya](https://openjdk.org/census#iwalulya) (@walulyai - **Reviewer**) ⚠️ Review applies to [efcc6289](https://git.openjdk.org/jdk/pull/10656/files/efcc628955319599712a38aefc12e13d2788b6d8)
 * [Albert Mingkun Yang](https://openjdk.org/census#ayang) (@albertnetymk - **Reviewer**) ⚠️ Review applies to [efcc6289](https://git.openjdk.org/jdk/pull/10656/files/efcc628955319599712a38aefc12e13d2788b6d8)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10656/head:pull/10656` \
`$ git checkout pull/10656`

Update a local copy of the PR: \
`$ git checkout pull/10656` \
`$ git pull https://git.openjdk.org/jdk pull/10656/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10656`

View PR using the GUI difftool: \
`$ git pr show -t 10656`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10656.diff">https://git.openjdk.org/jdk/pull/10656.diff</a>

</details>

### CODE REVIEW

**Review by walulyai — APPROVED** [review]
Lgtm!

File: src/hotspot/share/gc/g1/g1_globals.hpp [hunk 1]
```diff
@@ -377,6 +377,11 @@
           "retained region restore purposes.")                              \
           range(1, 256)                                                     \
                                                                             \
+  product(uint, G1NumCardsCostSampleThreshold, 1000, DIAGNOSTIC,            \
```
Comment (albertnetymk) [inline src/hotspot/share/gc/g1/g1_globals.hpp:380]: Unfounded hunch: an inclusive threshold here is probably easier to reason.

**Review by albertnetymk — APPROVED** [review]

File: src/hotspot/share/gc/g1/g1_globals.hpp (hunk 1 above)
Comment (tschatzl) [reply src/hotspot/share/gc/g1/g1_globals.hpp:380]: Either is fine with me. Changed.

**Comment by tschatzl** [comment]
Thanks @albertnetymk @walulyai for your reviews
/integrate

**Pull request closed without merging**